micro_sam/automatic_segmentation.py

```diff
@@ -478,37 +478,36 @@ def _convert_argval(value):
     )
 
     # Run automatic segmentation per image.
-    for input_path in tqdm(input_paths, desc="Run automatic segmentation"):
+    for i, input_path in tqdm(enumerate(input_paths), total=len(input_paths), desc="Run automatic segmentation"):
         if has_one_input:  # if we have only one image / volume.
             _embedding_fpath = embedding_path
-            if args.tracking:
-                _output_fpath = output_path
-            else:
-                _output_fpath = str(Path(output_path).with_suffix(".tif"))
+            _output_fpath = output_path if args.tracking else str(Path(output_path).with_suffix(".tif"))
 
         else:  # if we have multiple image, we need to make the other target filepaths compatible.
+
+            # Decide on the input filename.
+            input_name = f"embeddings_{i:05d}" if args.tracking else Path(input_path).stem
```

Comment:
Having the `embeddings` in input_name doesn't make sense, as you use it below as the name for the results.